### PR TITLE
UIU-576: Implement a Password Strength Meter

### PR DESCRIPTION
**Pull request by OleksandrAntonenko1**

### CODE REVIEW

**Comment by cherewaty** [comment]
We've got a ticket this week to set up the BigTest infrastructure in `stripes-smart-components`: https://issues.folio.org/browse/STSMACOM-122. It should simplify this PR quite a bit, and I think it will have to involve setting up the `stripes` context, which the `stripes-components` harness doesn't do. I'll keep you in the loop on that ticket.

**Comment by funkymalc** [comment]
Specify 'runTestOption' in the Jenkinsfile.   See stripes-components for an example.

File: lib/PasswordStrength/PasswordSrength.css [hunk 1]
```diff
@@ -0,0 +1,64 @@
+
```
Comment (VictorSoroka1) [inline lib/PasswordStrength/PasswordSrength.css]: Remove empty lines at the top.

File: lib/PasswordStrength/PasswordSrength.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+.password-strength-text__wrapper {
+  display: flex;
+  align-items: center;
+}
+
+.indicator__container {
+  display: flex;
+  justify-content: space-between;
+  flex-grow: 2;
+  margin: 6px 0 0 0;
```
Comment (VictorSoroka1) [inline lib/PasswordStrength/PasswordSrength.css]: Can we just use `margin-top`?

File: lib/PasswordStrength/PasswordStrength.js [hunk 3]
```diff
@@ -2,45 +2,102 @@ import React from 'react';
...
     return (
-      <div>
-        <FormattedMessage id={'stripes-smart-components.passwordStrength.' + this.output} />
-      </div>
+      <Row>
+        <Col xs={6} {...inputColProps}>
+          <TextField {...this.props} />
+        </Col>
+        <Col xs={4} {...passwordMeterColProps} aria-live="polite">
+          {valid && !hidden && dirty &&
```
Comment (VictorSoroka1) [inline lib/PasswordStrength/PasswordStrength.js]: Let's have `valid && !hidden && dirty` expression as its own variable for readability purpose.

File: lib/PasswordStrength/PasswordStrength.js [hunk 4]
```diff
@@ -2,45 +2,102 @@ import React from 'react';
...
+            <div className="password-strength">
+              <div className={css['password-strength__label']}>
+                <FormattedMessage
+                  id="stripes-smart-components.passwordStrength.label"
+                />
+              </div>
+              <div
+                className={`
+            ${css.indicator__container}
```
Comment (VictorSoroka1) [inline lib/PasswordStrength/PasswordStrength.js]: Indentation issue.

File: lib/PasswordStrength/PasswordStrength.js [hunk 5]
```diff
@@ -2,45 +2,102 @@ import React from 'react';
...
+    hidden: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
     this.strengthTester = new taiPasswordStrength.PasswordStrength();
     this.strengthTester.addCommonPasswords(taiPasswordStrength.commonPasswords);
     this.strengthTester.addTrigraphMap(taiPasswordStrength.trigraphs);
-    this.output = 'weak'; // set default password hint to weak
+    this.output = 'veryWeak'; // set default password hint to weak
```
Comment (VictorSoroka1) [inline lib/PasswordStrength/PasswordStrength.js]: // set default password hint to `veryWeak`?

File: lib/PasswordStrength/PasswordSrength.css (hunk 1 above)
Comment (OleksandrAntonenko1) [reply lib/PasswordStrength/PasswordSrength.css]: Done.

**Comment by cherewaty** [comment]
I think we should hold on adding any tests until https://issues.folio.org/browse/STSMACOM-122 is done. The configuration added here will work for this component, but others like `SearchAndSort` really need an entire `stripes` god object set up to be effectively tested.

My recommendation: could we split this into two pull requests?
1. the new parts of `PasswordStrength`
2. the test, interactor, and BigTest configuration

We could get part 1 merged much sooner than part 2.

File: package.json [hunk 6]
```diff
@@ -78,15 +79,36 @@
...
+    "react-trigger-change": "^1.0.2",
+    "karma": "^2.0.4",
+    "karma-junit-reporter": "^1.2.0",
+    "karma-webpack": "^3.0.0",
     "@folio/eslint-config-stripes": "^3.2.0",
     "babel-eslint": "^9.0.0",
+    "babel-preset-es2015": "^6.18.0",
+    "babel-preset-react": "^6.16.0",
+    "babel-preset-stage-0": "^6.16.0",
+    "babel-register": "^6.18.0",
```
Comment (cherewaty) [inline package.json]: I don't think we need any of the four `babel-*` dependencies listed here?

File: package.json [hunk 7]
```diff
@@ -78,15 +79,36 @@
...
     "eslint": "^5.5.0",
+    "stylelint": "^9.5.0",
+    "stylelint-config-standard": "^18.2.0",
```
Comment (cherewaty) [inline package.json]: Got out of alphabetical order

File: package.json [hunk 8]
```diff
@@ -78,15 +79,36 @@
...
+    "chai": "^4.0.2",
+    "chai-jquery": "^2.0.0",
+    "babel-core": "^6.26.3",
+    "babel-loader": "^7.0.0",
+    "babel-polyfill": "^6.26.0",
+    "jquery": "^3.3.1",
+    "react-trigger-change": "^1.0.2",
+    "karma": "^2.0.4",
+    "karma-junit-reporter": "^1.2.0",
+    "karma-webpack": "^3.0.0",
```
Comment (cherewaty) [inline package.json]: We might not need any package.json changes until the PR with tests?

File: Jenkinsfile [hunk 9]
```diff
@@ -3,5 +3,6 @@ buildNPM {
   publishModDescriptor = 'no'
   runLint = 'yes'
   runTest = 'yes'
+  runTestOptions = '--karma.singleRun --karma.browsers ChromeDocker --karma.reporters mocha junit --coverage'
```
Comment (cherewaty) [inline Jenkinsfile]: Won't need this line until the PR that includes tests.

**Comment by VictorSoroka1** [comment]
Please handle organization questions with designer and product owner about the look of the component.

**Pull request merged**